### DISCUSSION TITLE
Notebook templates

### DISCUSSION THREAD

**Discussion by ghost** [fonsp/Pluto.jl#1964] [Questions about using Pluto] [answered]
I have a few different notebooks with prefilled cells that include code and formatting. Rather than opening them every time and then renaming them, I would like to put together a small package to do this, namely [PlutoTemplates.jl[(). I gave it a shot but I'm quickly finding out I'm not sure how to do this cleanly. I just want upon `using PlutoTemplates` to inject a dropdown menu, select my template style, and then inject the relevant Pluto cells and markdown/html/code. Once the template is loaded, modifications should obviously be persistent and upon reopening the notebook in Pluto shouldn't repopulate the template.

Just curious if anyone else is thinking about this.

![image](https://user-images.githubusercontent.com/65369940/156433266-9dd58de6-17cd-469c-99ca-400f6b01e3d7.png)

**Comment by disberd** [comment, marked as answer]
I also wanted a way to inject cells into a notebook quite a while back. I do not think you can inject cells into the notebook simply during the loading phase of the package. A not so drastic workaround to that would be to define in your package an initialization function (that you have to call yourself inside a separate cell) that outputs some javascript code to do what you want.

You can add, populate and execute cells from the frontend using javascript, I do not think you can do that from the julia side as a package outside of Pluto.
